### ISSUE TITLE
HelpBrowser: empty Symbol breaks colorcoding

### ISSUE THREAD

**Issue by sensestage** [supercollider/supercollider#1574]
In the HelpBrowser a code like this:

```
code::
'';
//some other code
::
```

breaks colorcoding.

Example in the LID:findBy method documentation.

**Comment by snappizz** [comment]
I'm not seeing this bug.

**Comment by vivid-synth** [comment]
@sensestage do you still see this?

**Comment by brianlheim** [comment]
Reproducer:
```
code::
LID.findBy( unique: '' )
LID.findBy( unique: "" ) // argument is converted to symbol for check
LID.findBy( unique: '' )
::
```
N.B. this is mostly fine, but the `''` isn't highlighted as it should be. What controls highlighting in the HelpBrowser?
```

code::
LID.findBy( unique: '' )
// extra code
::
```

**Comment by snappizz** [comment]
@brianlheim `HelpSource/lang-sc.js`

**Comment by brianlheim** [comment]
https://github.com/supercollider/supercollider/blob/master/HelpSource/lang-sc.js#L14:
```javascript
        [PR.PR_ATTRIB_NAME, /^'[^']+'/],
```
`+` should be a `*`. Will test later.

**Comment by brianlheim** [comment]
Okay, so this has actually expanded into a much larger undertaking. There is code in `/editors/sc-ide/core/sc_lexer.cpp` that defines the regexes for the IDE, which I think we should consider "the standard". Most of these don't match those `lang-sc.js` in pretty significant ways (for example, highlighting the - at the start of a number). I'd like to submit a PR that tackles all of this as well as the `fixme` for multi-line comments. I will come back to this when I have time.

**Comment by brianlheim** [comment]
Fixed in #3974, thanks @htor!
